mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -204,9 +204,10 @@ class GBTClassifier @Since("1.4.0") (
     val boostingStrategy = super.getOldBoostingStrategy(categoricalFeatures, OldAlgo.Classification)
     val (baseLearners, learnerWeights) = if (withValidation) {
       GradientBoostedTrees.runWithValidation(trainDataset, validationDataset, boostingStrategy,
-        $(seed), $(featureSubsetStrategy))
+        $(seed), $(featureSubsetStrategy), Some(instr))
```

Comment:
OK, it's slightly messy to pass the instrumentation object here, but I think none of the changes here require new computation. I'm OK with it if you think it's important to log.

Reply:
I hope it's slightly better now after I added the default value ```instr: Option[Instrumentation] = None```. The code here is still the same,  though.